src/Raven.Server/Documents/Handlers/IndexHandler.cs

```diff
@@ -102,6 +103,47 @@ public Task Source()
             return Task.CompletedTask;
         }
 
+        [RavenAction("/databases/*/indexes/history", "GET", AuthorizationStatus.ValidUser)]
+        public Task GetIndexHistory()
+        {
+            var name = GetQueryStringValueAndAssertIfSingleAndNotEmpty("name");
+            var record = Database.ReadDatabaseRecord();
+            List<IndexHistoryEntry> history;
+            if (record.IndexesHistory.TryGetValue(name, out history) == false || history == null || history.Count  == 0) 
+            {
+                throw new InvalidOperationException("Could not find an history for index: " + name);
```

Comment:
Let's return 404 here to match behavior from other endpoints.

Reply:
Let's only return 404 when index is not found. When history is null or count is 0 just return empty array.